bgpd/bgp_nht.c

```diff
@@ -1272,182 +1357,337 @@ void evaluate_paths(struct bgp_nexthop_cache *bnc)
 							  sizeof(bnc_buf)));
 	}
 
-	LIST_FOREACH (path, &(bnc->paths), nh_thread) {
-		if (path->type == ZEBRA_ROUTE_BGP &&
-		    (path->sub_type == BGP_ROUTE_NORMAL ||
-		     path->sub_type == BGP_ROUTE_STATIC ||
-		     path->sub_type == BGP_ROUTE_IMPORTED))
-			/* evaluate the path */
-			;
-		else if (path->sub_type == BGP_ROUTE_REDISTRIBUTE) {
-			/* evaluate the path for redistributed routes
-			 * except those from VNC
-			 */
-			if ((path->type == ZEBRA_ROUTE_VNC) ||
-			    (path->type == ZEBRA_ROUTE_VNC_DIRECT))
+	if (bnc->srte_color) {
+		LIST_FOREACH (path, &(bnc->paths), te_nh_thread) {
```

Comment:
Is this basically the same as `LIST_FOREACH (path, &(bnc->paths), nh_thread)`? Then can we just move to the separate function and reuse? Because now we have lots of duplicated stuff + the indentation is bad at some level (due to nesting).

Reply:
Sometimes, two BNC are needed, for example, one route has a color while the other route does not,  and they have the same nexthop, the route with a color select SRv6 TE policy, and the other route select SRv6 BE

Reply:
- for better readability, if you indent a huge part of code, the best would be to add a separate commit: put the code to indent in a separate function. Then, the next commit does the indentation with only the called function.